### PR TITLE
[prometheus-elasticsearch-exporter]: Bump chart api version, min kubernetes version, set standard labels

### PR DESCRIPTION
**Pull request by hlac9x**
#### What this PR does / why we need it
This Pull Request resolves an issue with duplicate release labels in the ServiceMonitor when using a custom release name for Elasticsearch outside the kube-prometheus-stack. Previously, if a release label was defined in the values.yaml file, it would be applied alongside the default Helm release name, leading to duplication. 

#### Which issue this PR fixes
This fix ensures that:

- Correct Release Label: The ServiceMonitor will only use the release label defined in the values.yaml file, if provided.
- Default Helm Release Name: If no custom release label is defined, the Helm release name will be used as the default, aligning with the standard behavior within the kube-prometheus-stack.
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

Further changes:
- set standard labels
- set chart api version to v2
- require Kubernetes version min 1.19

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by zeritti — COMMENTED** [review]
Thank you, @hlac9x, for your PR. I would like to suggest a bit more comprehensive change. 

This chart is one of a few that are still employing the old-style labels and selectors, e.g. `app`. Would you be willing to change to the current [standard labels](https://helm.sh/docs/chart_best_practices/labels/#standard-labels) and selectors, e.g. `app.kubernetes.io/name` by means of helper templates? An example is [alertmanager](https://github.com/prometheus-community/helm-charts/blob/18f399b1028084e7e924a4d1efe2ce39bbda1ee3/charts/alertmanager/templates/_helpers.tpl#L34).

This would be a major change, though, needing a note in README. We could at last move to chart v2 at the same time, taking advantage of the major change. What do you think?

**Comment by hlac9x** [comment]
> Thank you, @hlac9x, for your PR. I would like to suggest a bit more comprehensive change.
> 
> This chart is one of a few that are still employing the old-style labels and selectors, e.g. `app`. Would you be willing to change to the current [standard labels](https://helm.sh/docs/chart_best_practices/labels/#standard-labels) and selectors, e.g. `app.kubernetes.io/name` by means of helper templates? An example is [alertmanager](https://github.com/prometheus-community/helm-charts/blob/18f399b1028084e7e924a4d1efe2ce39bbda1ee3/charts/alertmanager/templates/_helpers.tpl#L34).
> 
> This would be a major change, though, needing a note in README. We could at last move to chart v2 at the same time, taking advantage of the major change. What do you think?

**Comment by hlac9x** [comment]
@zeritti I have update the PR. please help to review.

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @hlac9x, for your change. Please, see my comments below.

File: charts/prometheus-elasticsearch-exporter/values.yaml [hunk 1]
```diff
@@ -300,7 +302,8 @@ prometheusRule:
   enabled: false
   #  namespace: monitoring
   labels: {}
-  rules: []
+  rules:
+    []
```
[the remark this answers is not in the document]
Comment (hlac9x) [reply charts/prometheus-elasticsearch-exporter/values.yaml:306]: this was a mistake I made when testing the helm template

File: charts/prometheus-elasticsearch-exporter/values.yaml [hunk 2]
```diff
@@ -264,10 +265,11 @@ serviceMonitor:
   ## If true, a ServiceMonitor CRD is created for a prometheus operator
   ## https://github.com/coreos/prometheus-operator
   ##
-  enabled: false
+  enabled: true
```
[the remark this answers is not in the document]
Comment (hlac9x) [reply charts/prometheus-elasticsearch-exporter/values.yaml:268]: this was a mistake I made when testing the helm template

File: charts/prometheus-elasticsearch-exporter/values.yaml [hunk 3]
```diff
@@ -71,7 +71,8 @@ log:
   format: logfmt
   level: info
 
-resources: {}
+resources:
+  {}
```
[the remark this answers is not in the document]
Comment (hlac9x) [reply charts/prometheus-elasticsearch-exporter/values.yaml:75]: this was a mistake I made when testing the helm template

File: charts/prometheus-elasticsearch-exporter/values.yaml [hunk 4]
```diff
@@ -264,10 +265,11 @@ serviceMonitor:
   ## If true, a ServiceMonitor CRD is created for a prometheus operator
   ## https://github.com/coreos/prometheus-operator
   ##
-  enabled: false
+  enabled: true
   apiVersion: "monitoring.coreos.com/v1"
   #  namespace: monitoring
-  labels: {}
+  labels:
+    release: kube-prometheus-stack
```
[the remark this answers is not in the document]
Comment (hlac9x) [reply charts/prometheus-elasticsearch-exporter/values.yaml:271]: this was a mistake I made when testing the helm template

**Comment by hlac9x** [comment]
@zeritti Thank you for your suggestion; there was a mistake in the values. yaml file I created while testing the Helm template. Thank you for your corrections.

File: charts/prometheus-elasticsearch-exporter/values.yaml [hunk 5]
```diff
@@ -267,7 +267,8 @@ serviceMonitor:
   enabled: false
   apiVersion: "monitoring.coreos.com/v1"
   #  namespace: monitoring
-  labels: {}
+  labels:
+    release: kube-prometheus-stack
```
Comment (zeritti) [inline charts/prometheus-elasticsearch-exporter/values.yaml:271]: This label's value is not applicable to all users, can be included as a comment, though.
```suggestion
labels: {}
  # release: kube-prometheus-stack
```

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @hlac9x, for your changes. I will make some changes in README, NOTES and Chart.yaml if it is ok with you.

**Comment by hlac9x** [comment]
@zeritti yes, please go ahead

**Review by zeritti — COMMENTED** [review]
@hlac9x I have extended the upgrade note as users have to know what is to be done before they can upgrade. Due to changes of the labels, NOTES also needed an update of a label. The introduced commonLabels also apply to pods. At last, we are leaving chart api v1 and thus are supporting helm v2 no more.

**Review by zeritti — APPROVED** [review]

**Pull request merged**